src/Datadog.Trace.ClrProfiler.Native/clr_helpers.cpp

```diff
@@ -999,4 +1023,553 @@ bool ReturnTypeIsValueTypeOrGeneric(
           ret_type_token);
   }
 }
+
+
+// FunctionMethodArgument
+int FunctionMethodArgument::GetTypeFlags(unsigned& elementType) const {
+  int flag = 0;
+  PCCOR_SIGNATURE pbCur = &pbBase[offset];
+
+  if (*pbCur == ELEMENT_TYPE_VOID) {
+    elementType = ELEMENT_TYPE_VOID;
+    flag |= TypeFlagVoid;
+    return flag;
+  }
+
+  if (*pbCur == ELEMENT_TYPE_BYREF) {
+    pbCur++;
+    flag |= TypeFlagByRef;
+  }
+
+  elementType = *pbCur;
+
+  switch (*pbCur) {
+    case ELEMENT_TYPE_BOOLEAN:
+    case ELEMENT_TYPE_CHAR:
+    case ELEMENT_TYPE_I1:
+    case ELEMENT_TYPE_U1:
+    case ELEMENT_TYPE_U2:
+    case ELEMENT_TYPE_I2:
+    case ELEMENT_TYPE_I4:
+    case ELEMENT_TYPE_U4:
+    case ELEMENT_TYPE_I8:
+    case ELEMENT_TYPE_U8:
+    case ELEMENT_TYPE_R4:
+    case ELEMENT_TYPE_R8:
+    case ELEMENT_TYPE_I:
+    case ELEMENT_TYPE_U:
+    case ELEMENT_TYPE_VALUETYPE:
+    case ELEMENT_TYPE_MVAR:
+    case ELEMENT_TYPE_VAR:
+      flag |= TypeFlagBoxedType;
+      break;
+    case ELEMENT_TYPE_GENERICINST:
+      pbCur++;
+      if (*pbCur == ELEMENT_TYPE_VALUETYPE) {
+        flag |= TypeFlagBoxedType;
+      }
+      break;
+    default:
+      break;
+  }
+  return flag;
+}
+
+mdToken FunctionMethodArgument::GetTypeTok(ComPtr<IMetaDataEmit2>& pEmit, mdAssemblyRef corLibRef) const {
+  mdToken token = mdTokenNil;
+  PCCOR_SIGNATURE pbCur = &pbBase[offset];
+  const PCCOR_SIGNATURE pTemp = pbCur;
```

Comment:
nit: Can this be renamed to pStart?

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/1007/commits/146ad3b00fd779e612bb226d4c130b19003ee549